etna/datasets/tsdataset.py

```diff
@@ -165,18 +166,60 @@ def __repr__(self):
     def _repr_html_(self):
         return self.df._repr_html_()
 
-    def __getitem__(self, item):
-        if isinstance(item, slice) or isinstance(item, str):
-            df = self.df.loc[self.idx[item]]
-        elif len(item) == 2 and item[0] is Ellipsis:
-            df = self.df.loc[self.idx[:], self.idx[:, item[1]]]
-        elif len(item) == 2 and item[1] is Ellipsis:
-            df = self.df.loc[self.idx[item[0]]]
+    def __getitem__(self, item) -> "TSDataset":
+        def to_time_index(item):
+            int_slice = isinstance(item.start, int) or isinstance(item.stop, int)
+            if int_slice:
+                start = None if item.start is None else self.index[item.start]
+                stop = None if item.stop is None else self.index[item.stop]
+                item = slice(start, stop, item.step)
+            return item
+
+        df_exog = deepcopy(self.df_exog)
+
+        if isinstance(item, slice):
+            time_idx = to_time_index(item)
+            df = self.df.loc[self.idx[time_idx]]
+            raw_df = self.raw_df.loc[self.idx[time_idx]]
+        elif isinstance(item, tuple):
+            time_idx = to_time_index(item[0])
+            if len(item) == 2:
+                features = [item[1]] if isinstance(item[1], str) else item[1]
+                if not isinstance(features, slice):
+                    features = tuple(features)
+                df = self.df.loc[self.idx[time_idx], self.idx[:, features]]
+                raw_df = self.raw_df.loc[self.idx[time_idx]]
+            elif len(item) == 3:
+                segments = [item[1]] if isinstance(item[1], str) else item[1]
+                if not isinstance(segments, slice):
+                    segments = tuple(segments)
+                features = [item[2]] if isinstance(item[2], str) else item[2]
+                if not isinstance(features, slice):
+                    features = tuple(features)
+                df = self.df.loc[self.idx[time_idx], self.idx[segments, features]]
+                df_exog = None if df_exog is None else df_exog.loc[self.idx[:], self.idx[segments, :]]
+                raw_df = self.raw_df.loc[self.idx[time_idx], self.idx[segments, :]]
+            else:
+                raise ValueError("Unsupported indexing format!")
         else:
-            df = self.df.loc[self.idx[item[0]], self.idx[item[1], item[2]]]
+            raise ValueError("Unsupported indexing format!")
+
         first_valid_idx = df.first_valid_index()
```

Comment:
It seems this is not expected. 
We don't need it, because None at the begining make sense as for upstream TSDataset

Reply:
We do't need None at the begining, in the old version we also cut them in getitem

Reply:
Old version is not the target. It was developed in a hurry)

We cut them before, because this slices were used for persegment models mostly.

But it's important to note that there was and there is an idea that None at beginnig are about missed values of some process and we shouldn't just drop them without particular reason